search-app/src/cmr/search/services/query_walkers/facet_condition_resolver.clj

```diff
@@ -50,19 +64,20 @@
 
   (adjust-facet-query
    [c field-key]
+   ;; drop nested condition that has the facet field
    (when-not (has-field? (:condition c) field-key)
-     (update c :condition #(adjust-facet-query % field-key))))
+     c))
 
   cmr.common_app.services.search.query_model.StringCondition
   (has-field?
    [c field-key]
-   (if (re-matches (re-pattern (str field-key ".*")) (str (:field c)))
+   (if (string-condition-for-v2-facet-field? c field-key)
      true
      false))
 
   (adjust-facet-query
    [c field-key]
-   (when-not (re-matches (re-pattern (str field-key ".*")) (str (:field c)))
+   (when-not (string-condition-for-v2-facet-field? c field-key)
```

Comment:
Both of these (`re-matches` -> named condition) are sooo much better: easier to maintain, debug, understand, etc. Thanks.